feat: add workflow for updating specific snapshot

kstala opened the pull request:
<!--
Thank you for contributing to Shopware! Please fill out this description template to help us process your pull request.

Please make sure to fulfil our contribution guidelines (https://developer.shopware.com/docs/resources/guidelines/code/contribution?category=shopware-platform-dev-en/contribution).

Do your changes need to be documented?
Please create a second pull request at https://github.com/shopware/docs
-->

### 1. Why is this change necessary?
Currently, we can only update all snapshots at once (sometimes there are situations, such as valid issues, where there is no need to update all snapshots, only selected ones), we need an additional workflow that will update only the snapshots we select


### 2. What does this change do, exactly?


### 3. Describe each step to reproduce the issue or behaviour.


### 4. Please link to the relevant issues (if any).
<!-- Examples:
- closes #123  - closes the issue #123 when the PR is merged
- relates #123 - relates to the issue #123

If the issue exists only in Jira, include a link to the Jira issue.
- Jira issue: https://shopware.atlassian.net/browse/NEXT-123
-->

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/shopware/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfilled them

kstala committed:
feat: add workflow for updating specific snapshot

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/shopware?pullRequest=11734) <br/>All committers have signed the CLA.

philipreinken commented on .github/workflows/update-specific-snapshot.yml:
```diff
@@ -0,0 +1,30 @@
+        run: npx playwright install --with-deps
+      - name: Run Playwright to update specific snapshot
+        run: npx playwright test --grep "${{ github.event.inputs.testName }}" --update-snapshots
+      - name: Commit updated snapshot
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: 'Update snapshot ${{ github.event.inputs.snapshotName }}'
```
I'm not familiar with this action - will it just commit, or also create a new branch, push (+ open a PR)?
Because just committing in CI won't suffice I fear, as the changes will just be applied in the ephemeral runner repo.

kstala replied:
@philipreinken I tested it locally and on a private repo and it seems to just create a commit. What we want to achieve is simply updating specific snapshots, so this solution seemed to be the simplest to me. Maybe you know if it can be done in a better and more safe way?

kstala committed:
chore: replace commit action with creating PR

philipreinken committed:
chore(ci): add filter input for visual tests workflow

This allows us to limit execution, and e.g. snapshot re-generation to a specific set of tests defined by `filter`.

philipreinken commented on .github/workflows/visual-tests.yml:
@kstala WDYT?

Adding this parameter should allow us to limit execution to specific tests.
The other existing `update-snapshots` input, could be used in combination with this, leading to the snapshots for the branch being updated from just those specific tests that were selected.

Currently I don't think we need to open a PR for any of this, as the snapshots are [uploaded as artifacts](https://github.com/shopware/shopware/blob/trunk/.github/actions/ats/action.yaml#L180-L189), but I might be missing something, so please correct me if I'm mistaken here.

philipreinken replied:
Nevermind - @GitEvil just filled me in on how this is supposed to work.

I wasn't aware you guys were committing the snapshots to the repository - the [ATS action](https://github.com/shopware/shopware/blob/trunk/.github/actions/ats/action.yaml) currently only handles artifacts, so we'll need to adjust the base action for this workflow; I've opened a corresponding [issue](https://github.com/shopware/shopware/issues/12467) for this.

philipreinken commented:
I've implemented the change in: https://github.com/shopware/shopware/pull/12470
